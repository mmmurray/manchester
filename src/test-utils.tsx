import * as RTL from '@testing-library/react'
import React, { FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from './theme'

const Wrapper: FC = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children as any}</ThemeProvider>
)

const render = (el: React.ReactElement) => RTL.render(<Wrapper>{el}</Wrapper>)

export * from '@testing-library/react'
export { render, Wrapper }
