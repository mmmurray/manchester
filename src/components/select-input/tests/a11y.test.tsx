import React from 'react'
import SelectInput from '..'
import { Wrapper } from '../../../test-utils'

const options = [
  { name: 'A', value: 'a' },
  { name: 'B', value: 'b' },
  { name: 'C', value: 'c' },
]

test('default select input a11y', async () => {
  await expect(
    <Wrapper>
      <SelectInput label="test" options={options} />
    </Wrapper>,
  ).toMatchA11ySnapshot()
})
