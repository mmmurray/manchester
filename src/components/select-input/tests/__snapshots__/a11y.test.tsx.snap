// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default select input a11y: toMatchA11ySnapshot 1`] = `
"role: WebArea
children:
  -
    role: Label
    children:
      -
        role: text
        name: test
      -
        role: combobox
        name: test
        value: A
        children:
          -
            role: MenuListPopup
            children:
              -
                role: menuitem
                name: A
                selected: true
              -
                role: menuitem
                name: B
              -
                role: menuitem
                name: C
"
`;
